### ISSUE TITLE
Validator status codes cleanup

### ISSUE THREAD

**Issue by mratsim** [ethereum/eth2.0-specs#216]
Following https://github.com/ethereum/eth2.0-specs/pull/185 the Validator status code changed to:

| Name | Value |
| - | - |
| `PENDING_ACTIVATION` | `0` |
| `ACTIVE` | `1` |
| `EXITED_WITHOUT_PENALTY` | `2` |
| `EXITED_WITH_PENALTY` | `3` |

But there are 2 PENDING_EXIT leftover in the specs.

```python
def get_active_validator_indices(validators: [ValidatorRecords]) -> List[int]:
    """
    Gets indices of active validators from ``validators``.
    """
    return [i for i, v in enumerate(validators) if v.status in [ACTIVE, PENDING_EXIT]]
```

```python
def exit_validator(index: int,
                   state: BeaconState,
                   penalize: bool,
                   current_slot: int) -> None:
    """
    Exit the validator with the given `index`.
    Note that this function mutates `state`.
    """
    state.validator_registry_exit_count += 1

    validator = state.validator_registry[index]
    validator.latest_status_change_slot = current_slot
    validator.exit_count = state.validator_registry_exit_count

    # Remove validator from persistent committees
    for committee in state.persistent_committees:
        for i, validator_index in committee:
            if validator_index == index:
                committee.pop(i)
                break

    if penalize:
        validator.status = EXITED_WITH_PENALTY
        state.latest_penalized_exit_balances[current_slot // COLLECTIVE_PENALTY_CALCULATION_PERIOD] += get_effective_balance(validator)
        
        whistleblower = state.validator_registry[get_beacon_proposer_index(state, current_slot)]
        whistleblower_reward = validator.balance // WHISTLEBLOWER_REWARD_QUOTIENT
        whistleblower.balance += whistleblower_reward
        validator.balance -= whistleblower_reward
    else:
        validator.status = PENDING_EXIT # <--------------------------------

    state.validator_registry_delta_chain_tip = get_new_validator_registry_delta_chain_tip(
        validator_registry_delta_chain_tip=state.validator_registry_delta_chain_tip,
        index=index,
        pubkey=validator.pubkey,
        flag=EXIT,
    )
```

## Side-note

Also I notice that currently they are stored in a uint64. If we have less than 255 status code, we should use an uint8.

**Comment by hwwhww** [comment]
This issue is redundant to #195

> Also I notice that currently they are stored in a uint64. If we have less than 255 status code, we should use an uint8.

FYI It was discussed here: https://github.com/ethereum/eth2.0-specs/pull/170
But I understand your concern, do you mind opening an RFC for that?

**Comment by djrtwo** [comment]
Closing in favor of #195 
And currently being addressed in #217
